src/KOKKOS/sna_kokkos_impl.h

```diff
@@ -869,67 +921,90 @@ void SNAKokkos<DeviceType>::compute_bi(const typename Kokkos::TeamPolicy<DeviceT
   //            b(j1,j2,j) +=
   //              2*Conj(u(j,ma,mb))*z(j1,j2,j,ma,mb)
 
-  Kokkos::parallel_for(Kokkos::TeamThreadRange(team,idxb_max),
-      [&] (const int& jjb) {
-  //for(int jjb = 0; jjb < idxb_max; jjb++) {
-    const int j1 = idxb(jjb,0);
-    const int j2 = idxb(jjb,1);
-    const int j = idxb(jjb,2);
-
-    int jjz = idxz_block(j1,j2,j);
-    int jju = idxu_block[j];
-    double sumzu = 0.0;
-    double sumzu_temp = 0.0;
-    const int bound = (j+2)/2;
-    Kokkos::parallel_reduce(Kokkos::ThreadVectorRange(team,(j+1)*bound),
-        [&] (const int mbma, double& sum) {
-        //for(int mb = 0; 2*mb < j; mb++)
-          //for(int ma = 0; ma <= j; ma++) {
-        const int ma = mbma%(j+1);
-        const int mb = mbma/(j+1);
-        const int jju_index = jju+mb*(j+1)+ma;
-        const int jjz_index = jjz+mb*(j+1)+ma;
-        if (2*mb == j) return;
-        sum +=
-          ulisttot(jju_index,iatom).re * zlist(jjz_index,iatom).re +
-          ulisttot(jju_index,iatom).im * zlist(jjz_index,iatom).im;
-      },sumzu_temp); // end loop over ma, mb
-      sumzu += sumzu_temp;
-
-    // For j even, special treatment for middle column
-
-    if (j%2 == 0) {
-      const int mb = j/2;
-      Kokkos::parallel_reduce(Kokkos::ThreadVectorRange(team, mb),
-          [&] (const int ma, double& sum) {
-      //for(int ma = 0; ma < mb; ma++) {
-        const int jju_index = jju+(mb-1)*(j+1)+(j+1)+ma;
-        const int jjz_index = jjz+(mb-1)*(j+1)+(j+1)+ma;
-        sum +=
-          ulisttot(jju_index,iatom).re * zlist(jjz_index,iatom).re +
-          ulisttot(jju_index,iatom).im * zlist(jjz_index,iatom).im;
-      },sumzu_temp); // end loop over ma
-      sumzu += sumzu_temp;
-
-      const int ma = mb;
-      const int jju_index = jju+(mb-1)*(j+1)+(j+1)+ma;
-      const int jjz_index = jjz+(mb-1)*(j+1)+(j+1)+ma;
-      sumzu += 0.5*
-        (ulisttot(jju_index,iatom).re * zlist(jjz_index,iatom).re +
-         ulisttot(jju_index,iatom).im * zlist(jjz_index,iatom).im);
-    } // end if jeven
-
-    Kokkos::single(Kokkos::PerThread(team), [&] () {
-      sumzu *= 2.0;
-
-      // apply bzero shift
-
-      if (bzero_flag)
-        sumzu -= bzero[j];
-
-      blist(jjb,iatom) = sumzu;
-    });
-  });
+  int itriple = 0;
+  int idouble = 0;
+  int jalloy = 0;
+  for (int elem1 = 0; elem1 < nelements; elem1++)
+    for (int elem2 = 0; elem2 < nelements; elem2++) {
+      jalloy = idouble*idxz_max;
+      for (int elem3 = 0; elem3 < nelements; elem3++) {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, idxb_max),
+                             [&](const int &jjb) {
```

Comment:
Can we reduce the indent level here?

Reply:
@charlessievers can you fix this and any other whitespace issues?

Reply:
Just to be clear, all you want is for me to reduce the whitespace in the for loops I created?

Reply:
Yes.

Reply:
Yeah, if the indent is too large then it makes it hard to read.

Reply:
@stanmoore1 Have a look at the new version and tell me if this is what you were thinking?